formbricks/formbricks
# fix: delay stacked card container height calculation

## What does this PR do?



[legact-react-docs](https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node)
[similar-stackoverflow-issue](https://stackoverflow.com/questions/60476155/is-it-safe-to-use-ref-current-as-useeffects-dependency-when-ref-points-to-a-dom)

Accessing ref inside useEffect hook is not a good idea because of the nature of useEffect. 
Trying to access `const currentElement = cardRefs.current[questionIdxTemp];` returns null, thus height is set incorrect. 
As a solution, delaying this part a pit allows us the access rendered content.

Fixes #2784 



## How should this be tested?



- Go create survey with 3 questions
- First question is __Rating__ with long description
- Second question is __Free Text__ 
- Third question is __Free Text__
- Add logical so that when rating is 2 we show 2nd question and when rating is 3 we show 3rd question
- First give it a 3 rating and then click back
- Now give it a 2 rating and observe
- If doesn't work go back and give it a 3 rating again and observe

## Checklist



### Required


### Appreciated
